### ISSUE TITLE
Did you get the idea of UltimateKEYS from EurKEY?

### ISSUE THREAD

**Issue by zyxir** [pieter-degroote/UltimateKEYS#4]
Hi, Pieter,

Thank you for making UltimateKEYS, which saved my ass!

As a Chinese typer, I have always been upset at the fact that ALL the Chinese input method are based on the US QWERTY layout.

For most Chinese input methods, you toggle the input state between EN and CN with the Shift key. When in CN state, you can type Chinese characters with pinyin; and when in EN state, you can type latin letters and punctuations with the US QWERTY layout.

However, I am very interested at European languages like Spanish and French, which have special latin letters like "ç" and "á". If I want input these special letters while typing Chinese, I have to toggle the keyboard layout with Win+Space key, instead of toggle EN/CN state with Shift key, which is unbearable. Thus, I have been searching for a way to input special European letters efficiently while keeping the ability of typing Chinese for a long time.

I used to be a user of EurKEY by Steffen Bruentjen, while I was using Linux, because on Linux, EurKEY and Chinese input methods (like Fcitx) works together flawlessly.

However, not long ago I switched to Windows for some reason. While on Windows, I found that NONE of the Chinese input method support the EurKEY layout. ALL of the Chinese input methods uses the US QWERTY layout in EN state. If I wanted to type special European letters, I had to switch keyboard layout with Win+Space. I was quite upset at this. Afterwards, I found that **if I implement the EurKEY layout with AutoHotKey, it will work together with any Chinese input method.** Then implement EurKEY with AutoHotKey became my target.

While I was trying to implete the EurKEY layout on my own with AutoHotKey, I found your UltimateKEYS repository. What a fantastic project! UltimateKEYS supports more unicode characters than EurKEY, and UltimateKEYS is more nmemonic and natural. Most importantly, UltimateKEYS is implemented with AutoHotKey on Windows, so it can completely solves my problem of inputing European characters and Chinese characters simultaneously. Thank you, Pieter, for making this great piece of software!

Oh, and my question. Did you get the idea of UltimateKEYS from EurKEY?

**Comment by pieter-degroote** [comment]
Yes, indeed, it is also mentioned on my site (both projects are licensed as GPL V3) :

". . . **This project has adopted several aspects from EurKEY (by Steffen Brüntjen).**"
See :  [https://pieter-degroote.github.io/UltimateKEYS/](https://pieter-degroote.github.io/UltimateKEYS/)

(Experimental) Linux support is also available for UltimateKEYS and UltimateKEYS SW.

**Comment by pieter-degroote** [comment]
Compared to EurKEY, I tried to avoid some homoglyphs like :

° :  **degree sign - on the basic layout with Right Alt**
º :  masculine ordinal indicator (Spanish, Portuguese, Italian, Galician) - on a dead key combination
˚ :  ring above - only with a letter :  å/Å, ů/Ů, ẘ, ẙ

This to avoid putting the wrong character for °C and °F, etc.
I also tried puzzling somewhat to make the most interesting symbols directly available... For me, this tend to be a 'work in progress'.

To make typing for French, or French loanwords even 'swifter', recent releases also feature "Smart error-tolerant optimization for circumflex in French".
